keyvi/tests/keyvi/vector/basic_test.cpp

```diff
@@ -0,0 +1,144 @@
+/* * keyvi - A key value store.
+ *
+ * Copyright 2018   Narek Gharibyan<narekgharibyan@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  basic_test.h
+ *
+ *  Created on: March 17, 2018
+ *      Author: Narek Gharibyan <narekgharibyan@gmail.com>
+ */
+
+#include <exception>
+#include <string>
+
+#include <boost/test/unit_test.hpp>
+
+#include "vector/vector_types.h"
+
+namespace keyvi {
+namespace vector {
+
+BOOST_AUTO_TEST_SUITE(VectorTests)
+
+BOOST_AUTO_TEST_CASE(string_test) {
+  StringVectorGenerator generator;
+
+  const std::string filename = "vector_string.kvv";
+  const size_t size = 100;
+
+  for (size_t i = 0; i < size; ++i) {
+    generator.PushBack(std::to_string(i));
+  }
+
+  generator.WriteToFile(filename);
```

Comment:
would be good to cleanup files afterwards, even better if you use TMP, e.g. something like:

```
boost::filesystem::path temp_path = boost::filesystem::temp_directory_path();

temp_path /= boost::filesystem::unique_path("vector-unit-test-string-test-%%%%-%%%%-%%%%-%%%%");
std::string file_name = temp_path.string();

...

std::remove(file_name)
```